### ISSUE TITLE
No selection highlighting

### ISSUE THREAD

**Issue by mtreca** [bdarcus/bibtex-actions#47]
As mentioned in #45, selectrum's syntax highlighting is not shown when using `bibtex-action` functions. Looking at the README, there doesn't seem to be any highlighting either (except an current line background highlight).

![2021-03-19-180546_1920x1080_scrot](https://user-images.githubusercontent.com/16440823/112360948-2748c280-8cd3-11eb-9ff8-bc460664a82b.png)

I originally thought that changing `bibtex-completion-display-formats` was responsible (through `bibtex-actions-rich-ui`), but setting it to nil doesn't change anything.

The other major change from my implementation (which still displays faces properly) is the use of `propertize` in `bibtex-actions-get-candidates`, which I suspect resets pre-existing face settings in the completion minibuffer.

**Comment by bdarcus** [comment]
> The other major change from my implementation (which still displays faces properly) is the use of `propertize` in `bibtex-actions-get-candidates`, which I suspect resets pre-existing face settings in the completion minibuffer.

I think you're right.

``` elisp
ELISP> (propertize "test" 'face 'bar)
#("test" 0 4
  (face bar))

ELISP> (propertize (propertize "test" 'face 'bar) 'face 'bar2)
#("test" 0 4
  (face bar2))
```

Any ideas how to fix that?

**Comment by bdarcus** [comment]
Wait ... isn't this working correctly for me? This is with prelude.

![Screenshot from 2021-03-26 07-57-48](https://user-images.githubusercontent.com/1134/112628296-169a6880-8e09-11eb-8220-289b08128dae.png)

**Comment by bdarcus** [comment]
As we wait on a reply, this isn't technically a bug. Type "edi" with this.

``` elisp
(completing-read ";) "
                 (list (propertize "vim evil mode editor" 'display "vim editor")
                       (propertize "emacs best editor" 'display "emacs editor")))
```

I don't even think it's a bug in selectrum; just a question whether they are willing to support it. I agree it would be nice to have.

**Comment by bdarcus** [comment]
Daniel has been explaining the options. You can read that thread to understand.

**Comment by bdarcus** [comment]
Brainstorming:

What if:

1. the candidate string itself is generated from `bibtex-completion-format-entry`, which means that content will have the match highlighted
2. put the additional metadata important to search on in an "invisible" property; maybe this is also generated by a template? This part won't be highlighted, of course.

WDYT @mtreca?

**Comment by mtreca** [comment]
Hi @bdarcus, sorry, I am quite busy with work at the moment, so it might take me a while to anwser.

I read the thread you linked, and I think I agree with minad when stating that this trade-off is not great for the end user who will expect highlighting to work similarly everywhere, including bibtex-actions. Hence, I think using `bibtex-completion-format-entry` (which, if I understood correctly, would fix the highlighting issue) is preferable, especially if it makes it easier to potentially merge back in `bibtex-completion` later on (I don't know if you intend to do that).

Wouldn't the best answer be to make the upstream `bibtex-completion-format-entry` configurable, which would make it easier to extend on the original package. Is that what your open PR upstream aims at doing?

**Comment by bdarcus** [comment]
> Wouldn't the best answer be to make the upstream `bibtex-completion-format-entry` configurable, which would make it easier to extend on the original package. Is that what your open PR upstream aims at doing?

Yes.

I think I can figure it out. PR now opened.
